CsharpExtras/Dictionary/MultiValueMapImpl.cs

```diff
@@ -168,5 +154,13 @@ public override bool Equals(object obj)
             }
             return true;
         }
+
+        public override int GetHashCode()

```

Comment:
This algorithm isn't immediately clear. Is it a standard practice? Can we add some commenting around it to explain what's going on?